### ISSUE TITLE
Update WordCamp mu-plugins to work with Pilot Sites

### ISSUE THREAD

**Issue by iandunn** [WordPress/wordcamp.org#906]
Similar to #886, the `mu-plugins` and `mu-plugins-private` folders will need to be audited and updated to work with the Pilot network.

If there's time, some of the things in `mu-plugins-private` could be moved into the public `mu-plugins` folder, per #576 .

## general checks

- [x] #920
- [x] #936 
- [x] Move mu-plugins that should only run on the WordCamp network into `mu-plugins/wordcamp`, and add a `require` for them. Sometimes this may mean only moving certain functions from a file, rather than the entire file.
- [x] Check if any instances of `SITE_ID_CURRENT_SITE` should to be updated to use `WORDCAMP_NETWORK_ID` or `EVENTS_NETWORK_ID`. The latter are more explicit/descriptive if we want the code to only use one of the networks.
- [ ] #938 
- [ ] check all `(get|update|delete)_site_option` calls and see if we want to share that data across networks (based on context). if we do, then they can be updated to use `{action}_network_option( WORDCAMP_NETWORK_ID, ...)` instead. add a comment documenting the reason they're shared
- [ ] Test each plugin to make sure it works on both networks and doesn't have any unintended consequences

## individual plugins

- [x] #915 
- [x] Move Let's Encrypt helper in #921 
- [x] #922
- [x] Re-enable `latest-site-hints.php` when tested/updated
- [x] Re-enable `load-other-mu-plugins.php` when tested/updated
- [x] SSO and GDPR should be permanently enabled. related cb577f071fa313c235823b1703d34fc953e1b16b. GDPR done in 3f48d5169848a3a24b0c1f0b61dac02ef74b2c94
- [x] Once it's all done, reorganize `load-other-mu-plugins.php` if needed. Some things may be `require`d twice because of the temporary code.

**Comment by outdoor2kode** [comment]
Is the 866 the issue you'd like to refer to here? As I'm a bit not sure how to connect with this issue, maybe I missed some stuff, or perhaps it's issue 886? (not linking them here to prevent them from being referenced in both tickets)

**Comment by dd32** [comment]
@iandunn In r4453-wordcamp (plus https://github.com/WordPress/wordcamp.org/commit/cb577f071fa313c235823b1703d34fc953e1b16b as I didn't realise it was a git tracked file) and r20546-dotorg I've enabled some of the mu-plugins in `load-other-mu-plugins.php` for production authentication to succeed, and GDPR erasure/exports to stop failing.

Not the prettiest, but seemed like the easiest option forward, without enabling the jetpack-tweaks and other unreviewed mu-plugins.
Notes:
 1. The `*.WordPress.org` cookies are used for auth, which necessitated using SSO and the same auth keys.
 2. I think I could have made it work by setting WordCamp cookies on `events.wordpress.org` in addition to wporg cookies, but this has potential conflicts around logging in/out of multiple accounts, and multiple assumptions in the SSO code that `*.wordpress.org` is always using those cookies.
 3. The `nonce` and other salts are switched to use WordPress.org ones, I was unsure if these are used for any stored data in other plugins. I suspected not though.

If you have strong opinions on the reverse direction, I can _probably_ make `2` work, especially if `3` is a concern.
